build/core/build_center.py

```diff
@@ -65,7 +66,14 @@ def construct_graph(self):
             service_name = service_name[-2] if len(service_name) > 1 else None
             for file_name in file_list:
                 if file_name.endswith(".dockerfile"):
-                    self.graph.add_image_to_service(str(os.path.splitext(file_name)[0]), service_name)
+                    passed = False
+                    if self.task_type == 'all':
+                        passed = True
+                    elif file_name.endswith(".common.dockerfile") or file_name.endswith(
+                            ".{0}.dockerfile".format(self.task_type)):
```

Comment:
why not determine by `service_type` in config?

Reply:
Build should be seperated from deploy. And some service don't have image (/build folder). And some services and its build folder are not in the same path or their name is different (yarn-exporter and hadoop-resource-manager).